polkadot-protocol/smart-contracts-basics/evm-vs-polkavm.md

```diff
@@ -0,0 +1,69 @@
+
+Resource limiting in cross-contract calls serves a critical security purpose, particularly when interacting with untrusted contracts.
+
+For compatibility, PolkaVM maps traditional gas-related operations to its `ref_time` metric - the closest analog to Ethereum's gas system.
+
+## Memory Management
+
+The EVM and the PolkaVM take fundamentally different approaches to memory constraints:
+
+- **EVM's approach** - Ethereum uses gas costs as an indirect memory control mechanism. Beyond a 24KB code size limit, memory usage is constrained through a gas curve that increases costs based on total allocation. While elegant, this approach can lead to overcharging for memory operations, reducing overall system efficiency
+
+- **PolkaVM's approach** - PolkaVM takes a different approach to memory limits than Ethereum. Instead of variable gas costs, it implements fixed-cost operations with hard memory limits. This separation of memory constraints from execution time gas enables more efficient pricing, though it can occasionally result in more restrictive [limits](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/revive/src/limits.rs){target=\_blank}. The system's parameters are continuously adjusted to optimize functionality based on real-world usage patterns and community feedback
+
+    The architecture establishes a constant memory limit per contract, which is the basis for calculating maximum contract nesting depth. This calculation assumes worst-case memory usage for each nested contract, resulting in a straightforward but conservative limit that operates independently of actual memory consumption. Future iterations may introduce dynamic memory metering, allowing deeper nesting depths for contracts with smaller memory footprints. However, such an enhancement would require careful implementation of cross-contract boundary limits before API stabilization, as it would introduce an additional resource metric to the system.
+
+## Account Management - Existential Deposit
+
+Polkadot implements an [existential deposit](http://localhost:8000/polkadot-protocol/glossary/#existential-deposit){target=\_blank} (ED) system, requiring accounts to maintain a minimum balance to exist. When an account's balance falls below this threshold, the account is automatically deleted. This mechanism prevents state bloat from inactive accounts, unlike Ethereum, where accounts persist indefinitely regardless of balance and require no minimum funds to maintain their associated data structures (such as the nonce). This requirement extends to smart contracts, which operate as specialized accounts containing executable code.
```

Comment:
I like the EVM approach and Polkadot approach bullet points you used in the previous section. It might be nice to do something similar here, so users can look quickly and get the details

Reply:
Can you please check and see where else we could apply that type of formatting in the guide?